### ISSUE TITLE
Always launch in JIT mode on iOS 14.2 even when debugger isn't attached (remove ptrace check)

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#69984]
iOS 14.2 removed the requirement for a debugger to be attached to use JIT compilation (or to have a private entitlement).

Remove the ptrace check and `In iOS 14+, debug mode Flutter apps can only be launched from Flutter tooling` warning text.

https://github.com/utmapp/UTM/issues/397#issuecomment-719807123
https://9to5mac.com/2020/11/06/ios-14-2-brings-jit-compilation-support-which-enables-emulation-apps-at-full-performance/

TL;DR Allow apps to be launched from the home screen again.

**Comment by jmagman** [comment]
Make [`EnableTracingIfNecessaryOnce`](https://github.com/flutter/engine/blob/f0fb74b8fb52909e5bce45e176b00db1f0acae9e/runtime/ptrace_check.cc#L117-L127) return true on this version.

**Comment by saagarjha** [comment]
Stopping by to mention https://github.com/utmapp/UTM/issues/516 as something that you might want to watch.

**Comment by jmagman** [comment]
@saagarjha Thanks for the heads up. I have it working on an A12 running 18B92, I wish I had an A11 to test with...

**Comment by jmagman** [comment]
@xster tried on an iPhone 6s plus (A9) and hit `SIGKILL - CODESIGNING` 😞

**Comment by jmagman** [comment]
I don't want the engine to maintain an allowlist of model numbers.

I'm not endorsing Flutter weak linking on ARKit, but [`ARGeoTrackingConfiguration.isSupported`](https://developer.apple.com/documentation/arkit/argeotrackingconfiguration) and [`ARWorldTrackingConfiguration.supportsFrameSemantics(.personSegmentationWithDepth)`](https://developer.apple.com/documentation/arkit/occluding_virtual_content_with_people#3402799) are only true on A12 and higher... Just sayin'.

Maybe there's a more legit way to detect this?

**Comment by saagarjha** [comment]
I would guess that this is gated on arm64e support, but more digging into the kernel would be needed to figure out which models are currently supported.

**Comment by jmagman** [comment]
> I would guess that this is gated on arm64e support

```c++
  cpu_type_t cputype = 0;
  if (::sysctlbyname("hw.cputype", &cputype, &cputype_size, nullptr, 0) < 0) {
...
  }

  cpu_subtype_t cpusubtype = 0;
  if (::sysctlbyname("hw.cpusubtype", &cpusubtype, &cputype_size, nullptr, 0) < 0) {
...
  }

  return (cputype == CPU_TYPE_ARM64 && cpusubtype == CPU_SUBTYPE_ARM64E);
```
Confirmed I'm getting `CPU_SUBTYPE_ARM64E` on my A11 and `CPU_SUBTYPE_ARM64_V8` on my A9.

**Comment by jmagman** [comment]
Speculative PR at https://github.com/flutter/engine/pull/22377.  Would be nice to get confirmation on an A13 (iPhone 11) or A14 (iPhone 12).

**Comment by jamesblasco** [comment]
Some bad news related to JIT on ios  https://github.com/utmapp/UTM/issues/2271#issuecomment-761725159

**Comment by jmagman** [comment]
@jamesblasco thanks for the update; bad news indeed. 

> It appears that JIT worked < iOS 14.4 due to a bug/oversight. In iOS 14.4 beta 2, the application needs to be attached to a debugger for code signing to be disabled (just like < iOS 14.2). Since the ptrace() trick was also fixed, it seems unlikely that JIT will work on non-jailbroken devices (untethered) anytime soon. So don't update past iOS 14.3 if you care about JIT apps.
 
I wonder where they learned "JIT worked < iOS 14.4 due to a bug/oversight" and not that it stopped working due to a bug/oversight.

@zanderso suggested we not do this until we get more info https://github.com/flutter/engine/pull/22377#issuecomment-745526639
> I think we should probably abandon until we can get some more solid info about the devices where this is supported.

**Comment by jmagman** [comment]
Closing, Apple seems to intend to keep this behavior.
